### ISSUE TITLE
BlobDB in RocksDB cann't write to the spdk filesystem

### ISSUE THREAD

**Issue by 979357361** [spdk/spdk#2745]
# Sighting report

I run rocksdb on spdk blobfs, find that if rocksdb opens blobdb, the blobdb will write to the default path (/tmp/...), different from the rocksdb's LSM tree path (blobfs disk).

As far as I know, rocksdb doesn't have a separate api to configure blobdb location -- by default, blobdb is always stored in the same directory as LSM tree, but on spdk, things changed.

## Expected Behavior

Blobdb should stay with the LSM tree, which is on the spdk's blobfs (spdk bdev). Only then can rocksdb enjoy the benefits of spdk

## Current Behavior

The separate with each other -- LSM Tree on the spdk bdev and blobdb on the default device.

## Context (Environment including OS version, SPDK version, etc.)

RocksDB: 6.15.fb
SPDK: the latest
OS: Centos7

**Comment by changpe1** [comment]
@979357361 you can refer to `test/blobfs/rocksdb/rocksdb.sh` for a comparison, all SST files will be stored into SPDK' owned directory, I guess there is something wrong in your configuration parameters.

**Comment by 979357361** [comment]
> @979357361 you can refer to `test/blobfs/rocksdb/rocksdb.sh` for a comparison, all SST files will be stored into SPDK' owned directory, I guess there is something wrong in your configuration parameters.

@changpe1  for SST file, that's true, but not for *.blob file used by rocksdb blobdb. 
I search in the source code of Rocksdb-6.15-fb, find that the LSM tree is configured to used the blobfs provided by SPDK, but for blobdb, the dir paramater is "string blob_file_paths" and it is left as default.

**Comment by changpe1** [comment]
> > @979357361 you can refer to `test/blobfs/rocksdb/rocksdb.sh` for a comparison, all SST files will be stored into SPDK' owned directory, I guess there is something wrong in your configuration parameters.
> 
> @changpe1 for SST file, that's true, but not for *.blob file used by rocksdb blobdb. I search in the source code of Rocksdb-6.15-fb, find that the LSM tree is configured to used the blobfs provided by SPDK, but for blobdb, the dir paramater is "string blob_file_paths" and it is left as default.

I see, will you continue to investigate this issue ?

**Comment by 979357361** [comment]
> I see, will you continue to investigate this issue ?

yes, since kv separation in rocksdb means a lot if the value is gaint.

**Comment by jimharris** [comment]
[Bug Scrub] BlobDB came out well after the RocksDB SPDK plugin was developed.  Clearly more investigation is required to determine what's needed to support BlobDB with the SPDK plugin.  Marking as an enhancement request for now, not a bug.
